### PR TITLE
feat: Add UploadSuccess screens

### PR DESCRIPTION
**Pull request by KevinBoulongne**
Depends on #102

### CODE REVIEW

**Review by LunarX — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/upload/UploadSuccessQrScreen.kt [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+import com.infomaniak.swisstransfer.ui.screen.newtransfer.importfiles.components.TransferType
+import com.infomaniak.swisstransfer.ui.theme.Dimens
+import com.infomaniak.swisstransfer.ui.theme.Margin
+import com.infomaniak.swisstransfer.ui.theme.SwissTransferTheme
+import com.infomaniak.swisstransfer.ui.utils.PreviewAllWindows
+
+@Composable
+fun UploadSuccessQrScreen(transferType: TransferType) {
+
+    val uploadSuccessTitle = if (transferType == TransferType.QR_CODE) {
```
Comment (tevincent) [inline app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/upload/UploadSuccessQrScreen.kt:45]: That's weird to have an `UploadSuccessQrScreen` with a `transferType` different from `TransferType.QR_CODE`. Wouldn't it be `UploadSuccessScreen` ? Or something similar ? To handle both cases ? Or even if `transferType != TransferType.QR_CODE`, we show the QRCode ?

File: app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/upload/UploadSuccessQrScreen.kt [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+                Text(
+                    text = stringResource(uploadSuccessTitle),
+                    style = SwissTransferTheme.typography.h1,
+                    color = SwissTransferTheme.colors.primaryTextColor,
+                )
+
+                Spacer(Modifier.height(Margin.XLarge))
+
+                QrCode()
```
Comment (tevincent) [inline app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/upload/UploadSuccessQrScreen.kt:93]: Do you have to show this if `transferType != TransferType.QR_CODE`

**Review by tevincent — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/upload/UploadSuccessEmailScreen.kt [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        content = {
+            Column(
+                modifier = Modifier
+                    .fillMaxSize()
+                    .verticalScroll(rememberScrollState()),
+                horizontalAlignment = Alignment.CenterHorizontally,
+                verticalArrangement = Arrangement.Center,
+            ) {
+
+                IllustratedMessageBlock(
```
[the remark this answers is not in the document]
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/upload/UploadSuccessEmailScreen.kt:55]: The content of this screen is in a Column, so I added spaces to better see what's there.

**Review by tevincent — APPROVED** [review]

**Review by LunarX — CHANGES_REQUESTED** [review]
Your branch doesn't build

**Review by LunarX — CHANGES_REQUESTED** [review]
First part of the comments

File: app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/upload/UploadSuccessQrScreen.kt [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+fun UploadSuccessQrScreen(transferType: TransferType) {
+    BottomStickyButtonScaffold(
+        topBar = { BrandTopAppBar() },
+        topButton = {
+            LargeButton(
+                modifier = it,
+                style = ButtonType.PRIMARY,
+                titleRes = R.string.buttonShare,
+                onClick = { /* TODO */ },
+            )
```
Comment (LunarX) [inline app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/upload/UploadSuccessQrScreen.kt:52]: You can add the share feature already, we're gonna need it. Maybe it could go in its own PR though

**Review by LunarX — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/upload/UploadSuccessEmailScreen.kt [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+        IllustratedMessageBlock(
+            icon = AppIllus.UploadSuccessEmail.image(),
+            title = R.string.uploadSuccessEmailTitle,
+            description = R.string.uploadSuccessEmailDescription,
+            modifier = Modifier.padding(horizontal = Margin.Medium),
+        )
+
+        Spacer(Modifier.height(Margin.Small))
+
+        EmailAddressChip("test.test@ik.me") // TODO: Use correct email instead of hard-coded value.
```
[the remark this answers is not in the document]
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/upload/UploadSuccessEmailScreen.kt:69]: Questions asked on Figma.

File: app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/upload/UploadSuccessQrScreen.kt (hunk 4 above)
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/upload/UploadSuccessQrScreen.kt]: Yes, it will be in another one.

File: app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/upload/UploadSuccessEmailScreen.kt (hunk 5 above)
[the remark this answers is not in the document]
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/upload/UploadSuccessEmailScreen.kt]: Question answered !

We probably want to use a LazyStaggeredGrid.

**Review by LunarX — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/upload/UploadSuccessEmailScreen.kt [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+            icon = AppIllus.UploadSuccessEmail.image(),
+            title = R.string.uploadSuccessEmailTitle,
+            description = R.string.uploadSuccessEmailDescription,
+            modifier = Modifier.padding(horizontal = Margin.Medium),
+        )
+
+        FlowRow(
+            modifier = Modifier.padding(Margin.Medium),
+            horizontalArrangement = Arrangement.Center,
+        ) {
```
Comment (LunarX) [inline app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/upload/UploadSuccessEmailScreen.kt:89]: Add a max width so the chips don't stretch to infinity on large screens like tablets. Maybe something like 600, we need to check what's visually appealing

**Review by LunarX — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/upload/UploadSuccessEmailScreen.kt [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+            emails.forEach {
+                EmailAddressChip(
+                    text = it,
+                    modifier = Modifier.padding(horizontal = Margin.XSmall)
+                )
+            }
+        }
+    }
+}
```
Comment (LunarX) [inline app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/upload/UploadSuccessEmailScreen.kt:98]: I think we should put the horizontal padding on the parent column and have a spacer to separate the IllustratedMessageBlock from the FlowRow instead of specifying padding on both IllustratedMessageBlock and FlowRow

**Review by LunarX — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/upload/UploadSuccessEmailScreen.kt (hunk 6 above)
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/upload/UploadSuccessEmailScreen.kt]: UI is pretty neat as it is.

File: app/src/main/java/com/infomaniak/swisstransfer/ui/components/IllustratedMessageBlock.kt [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+@Preview(name = "Light mode")
+@Preview(name = "Dark mode", uiMode = Configuration.UI_MODE_NIGHT_YES or Configuration.UI_MODE_TYPE_NORMAL)
+@Composable
+private fun IllustratedMessageBlockPreview() {
+    SwissTransferTheme {
+        Surface {
+            IllustratedMessageBlock(
+                icon = AppIllus.MascotSearching,
+                title = R.string.noTransferReceivedTitle,
+                description = "All transfers you receive over the last 30 days will appear here.",
```
Comment (LunarX) [inline app/src/main/java/com/infomaniak/swisstransfer/ui/components/IllustratedMessageBlock.kt:85]: I didn't spot that one earlier but you can still call 
```suggestion
                description = stringResource(R.string.noTransferReceivedDescription),
```

**Review by LunarX — CHANGES_REQUESTED** [review]

**Review by LunarX — APPROVED** [review]
